CONTRIBUTING.md

```diff
@@ -109,42 +115,346 @@ Check out [building instructions](https://github.com/prestodb/presto/tree/master
+        1. All new language features, new functions, and major features have documentation added
+        1. Release notes following release note guidelines are added for user visible changes
+* For large features, discuss your design with relevant code owners before you start implementing it.
+
 
 ## Code Style
 
 We recommend you use IntelliJ as your IDE. The code style template for the project can be found in the [codestyle](https://github.com/airlift/codestyle) repository along with our general programming and Java guidelines. In addition to those you should also adhere to the following:
 
-* Alphabetize sections in the documentation source files (both in table of contents files and other regular documentation files). In general, alphabetize methods/variables/sections if such ordering already exists in the surrounding code.
-* When appropriate, use the Java 8 stream API. However, note that the stream implementation does not perform well so avoid using it in inner loops or otherwise performance sensitive sections.
-* Categorize errors when throwing exceptions. For example, PrestoException takes an error code as an argument, `PrestoException(HIVE_TOO_MANY_OPEN_PARTITIONS)`. This categorization lets you generate reports so you can monitor the frequency of various failures.
-* Ensure that all files have the appropriate license header; you can generate the license by running `mvn license:format`.
-* Consider using String formatting (printf style formatting using the Java `Formatter` class): `format("Session property %s is invalid: %s", name, value)` (note that `format()` should always be statically imported). Sometimes, if you only need to append something, consider using the `+` operator.
-* Avoid using the ternary operator except for trivial expressions.
-* Use an assertion from Airlift's `Assertions` class if there is one that covers your case rather than writing the assertion by hand. Over time we may move over to more fluent assertions like AssertJ.
-* When writing a Git commit message, follow these [guidelines](https://chris.beams.io/posts/git-commit/).
+* **Naming**
+    * Avoid abbreviations
+      E.g. positionCount instead of positionCnt
+    * Line width and spacing
+        * Lines should be within 180 chars. Intellij “Reformat code” does NOT enforce the line width, so you’ll need to adjust it yourself.
+        * Long function declarations exceeding 180 characters shall be broken down into multiple lines, one argument/parameter per line. A good example is like:
+
+      ```java
+          public ParquetPageSource(
+                  ParquetReader parquetReader,
+                  List<Type> types,
+                  List<Optional<Field>> fields,
+                  List<String> columnNames,
+                  RuntimeStats runtimeStats)
+      
+      ```
+
+        * Do not put the first parameter on the same line as the function name, or move the parentheses to different lines. Bad example:
```

Comment:
Nice work providing an example of incorrectly broken down code for the reader to compare their code to. No change requested here, this is another callout of good work.